src/LockstepApi.ts

```diff
@@ -162,20 +166,67 @@ export class LockstepApi {
     return this;
   }
 
+  /**
+   * Configures this Lockstep API client to use an application name
+   * 
+   * @param appName The Application Name to use for this API session
+   */
+  public withApplicationName(appName: string): LockstepApi {
+    this.appName = appName;
+    return this;
+    
+  }
+
+  /**
+   * Returns the currently selected application name 
+   */
+  public getAppName(): any {
+    return this.appName;
+  }
+
   /**
    * Construct headers for a request
    */
   private getHeaders(): any {
-    if (this.apiKey !== null) {
+    const hostName = _hostname();
+     if (hostName!==null) {
+       return {
+         
+         "SdkName": this.sdkName,
```

Comment:
Can we change this a bit so we have less repetition?  Maybe something like this:

```
const hostName = _hostname();
let headers = { 
     "SdkName": this.sdkName,
     "SdkVersion": this.version,
     "ApplicationName": this.appName, 
     "MachineName": hostName, 
};
if (this.bearerToken !== null) {
  headers["Authorization"] = `Bearer ${this.bearerToken}`;
} else if (this.apiKey !== null) {
  headers["ApiKey"] = this.apiKey;
}
```

Reply:
Yes.